app/Repositories/Summit/DoctrineSummitProposedScheduleRepository.php

```diff
@@ -0,0 +1,79 @@
+<?php namespace App\Repositories\Summit;
+/**
+ * Copyright 2023 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+use App\Models\Foundation\Summit\IPublishableEvent;
+use Doctrine\ORM\Query\Expr\Join;
+use Dotenv\Exception\ValidationException;
+use models\summit\ISummitProposedScheduleRepository;
+use App\Repositories\SilverStripeDoctrineRepository;
+use models\summit\SummitProposedSchedule;
+use models\summit\SummitProposedScheduleSummitEvent;
+
+/**
+ * Class DoctrineSummitProposedScheduleRepository
+ * @package App\Repositories\Summit
+ */
+final class DoctrineSummitProposedScheduleRepository
+    extends SilverStripeDoctrineRepository
+    implements ISummitProposedScheduleRepository
+{
+    /**
+     * @return string
+     */
+    protected function getBaseEntity()
+    {
+       return SummitProposedSchedule::class;
+    }
+
+    /**
+     *@inheritDoc
+     */
+    public function getBySourceAndSummitId(string $source, int $summit_id): array
```

Comment:
this shold be an unique result
source and id is a unique IDX constraint at db

Reply:
done